hydra_base/lib/users.py

```diff
@@ -248,19 +249,21 @@ def inc_failed_login_attempts(username, flush=True, **kwargs):
         raises:
             HydraError if the username does not exist
     """
-
     try:
         user_i = db.DBSession.query(User).filter(User.username == username).one()
     except NoResultFound:
-        raise HydraError(username)
+        raise HydraLoginUserNotFound("User '{}' does not exist".format(username))
 
     if user_i.failed_logins is not None:
         user_i.failed_logins = user_i.failed_logins + 1
     else:
         user_i.failed_logins = 1
 
+    log.info("[ inc_failed_login_attempts ] User {} has currently {} failed logins".format(username, user_i.failed_logins))
     if flush is True:
         db.DBSession.flush()
+        # This is needed
+        transaction.commit()
```

Comment:
See my previous comment. Do not commit this transaction. If it is needed, then it needs to be solved elsewhere. Why is it needed exactly?

Reply:
Without the commit, a following flush got locked forever and breaks the code flow with a db exception on the max-lock-timeout exceeded.

Reply:
has this been removed?